jetty-openid/src/main/java/org/eclipse/jetty/security/openid/JwtDecoder.java

```diff
@@ -0,0 +1,96 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security.openid;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Map;
+
+import org.eclipse.jetty.util.ajax.JSON;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+/**
+ * Used to decode the ID Token from the base64 encrypted JSON Web Token (JWT).
+ */
+public class JwtDecoder
+{
+    private static final Logger LOG = Log.getLogger(JwtDecoder.class);
+
+    /**
+     * Decodes a JSON Web Token (JWT) into a Map of claims.
+     * @param jwt the JWT to decode.
+     * @return the map of claims encoded in the JWT.
+     */
+    public static Map<String, Object> decode(String jwt)
+    {
+        if (LOG.isDebugEnabled())
+            LOG.debug("decode {}", jwt);
+
+        String[] sections = jwt.split("\\.");
+        if (sections.length != 3)
+            throw new IllegalArgumentException("JWT does not contain 3 sections");
+
+        Base64.Decoder decoder = Base64.getUrlDecoder();
```

Comment:
You can initialize this once as a static field on the class.

Reply:
If you make this a static field you need to know that it is shareable, but that is not immediately obvious without clicking through to the implementation.